ql/src/java/org/apache/hadoop/hive/ql/ddl/table/storage/compact/AlterTableCompactOperation.java

```diff
@@ -88,7 +92,8 @@ else if (desc.getPartitionSpec() != null) {
     CompactionRequest compactionRequest = new CompactionRequest(table.getDbName(), table.getTableName(),
         compactionTypeStr2ThriftType(desc.getCompactionType()));
 
-    compactionRequest.setPoolName(desc.getPoolName());
+    compactionRequest.setPoolName(ObjectUtils.defaultIfNull(desc.getPoolName(), 
```

Comment:
minor: could we please extract poolName into a local var for readability?

Reply:
Done